plugins/device/deviceutils/store-utils.go

```diff
@@ -174,3 +178,77 @@ func IsVgExist(vgname string) bool {
 	}
 	return false
 }
+
+// getDeviceAvailableSize gets the device size and vgName using device Path
+func getDeviceAvailableSize(devicePath string) (string, uint64, error) {
+	devices, err := GetDevices()
+	if err != nil {
+		return "", 0, err
+	}
+	deviceVgName := strings.Split(devicePath, "/")[2]
+	for _, d := range devices {
+		if d.VgName == deviceVgName {
+			return d.VgName, d.AvailableSize, nil
+		}
+	}
+	return "", 0, nil
+}
+
+// CheckForAvailableVgSize prepares a brickName to vgName mapping in order to use while expanding lvm
+// Also it checks for sufficient space available on devices of current node.
+func CheckForAvailableVgSize(expansionSize uint64, bricksInfo []brick.Brickinfo) (map[string]string, bool, error) {
+
+	// map of brickName to vgName. Used while extending lv
+	brickVgMapping := make(map[string]string)
+
+	// map of deviceName to required free size in that device
+	requiredDeviceSizeMap := make(map[string]uint64)
+
+	// Map device Name to required free space on the device
+	for _, b := range bricksInfo {
+		if _, ok := requiredDeviceSizeMap[b.MountInfo.DevicePath]; !ok {
+			requiredDeviceSizeMap[b.MountInfo.DevicePath] = expansionSize
+		} else {
+			requiredDeviceSizeMap[b.MountInfo.DevicePath] += expansionSize
+		}
+	}
+
+	// Check in the map prepared in last step by looking through devices names and device available size of bricks from current node.
+	for _, b := range bricksInfo {
+		// retrieve device available size by device Name and return the vgName and available device Size.
+		vgName, deviceSize, err := getDeviceAvailableSize(b.MountInfo.DevicePath)
+		if vgName == "" {
+			return map[string]string{}, false, errors.New("could not find Vg name")
```

Comment:
move this error to error package

Reply:
Don't  need it anymore if I am returning error from getDeviceAvailableSize().